virttest/libvirt_xml/base.py

```diff
@@ -233,6 +233,57 @@ def virt_xml_validate(filename, schema_name=None):
         cmdresult.stderr = cmdresult.stderr_text
         return cmdresult
 
+    def setup_attrs(self, **attrs):
+        """
+        Setup attributes of an xml object
+
+        :param attrs: dict-type attributes to be set
+
+        Example:
+            dimm_device_attrs = {
+                'mem_model': 'dimm',
+                'target': {'size': 524288, 'size_unit': 'KiB', 'node': 0},
+                'alias': {'name': 'dimm2'},
+                'address': {'attrs': {'type': 'dimm', 'slot': '2', 'base': '0x120000000'}}
+            }
+
+            dimm_device = memory.Memory()
+            dimm_device.setup_attrs(**dimm_device_attrs)
+
+        The content of dimm_device would be:
+            <?xml version='1.0' encoding='UTF-8'?>
+            <memory model="dimm">
+                <target>
+                    <size unit="KiB">524288</size><node>0</node>
+                </target>
+                <alias name="dimm2" />
+                <address base="0x120000000" slot="2" type="dimm" />
+            </memory>
+
+        """
+        reset_all = True if attrs.get('reset_all') is True else False
```

Comment:
Would you mind providing an example of reset_all usages? 
I faced a problem as below if set it as True in attrs. Am I wrong?
```
>>> dargs={"managed":"yes", "mode":"subsystem", "type":"pci",'reset_all': True,"source":{'untyped_address':{"bus":"0x3b","domain":"0x0000","function":"0x1", "slot":"0x00"}}}
>>> dargs.get('reset_all')
True
>>> hostdev2.setup_attrs(**dargs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/tmp/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/libvirt_xml/base.py", line 265, in setup_attrs
    raise AttributeError('Cannot set attribute "%s" to %s object.'
AttributeError: Cannot set attribute "reset_all" to <class 'virttest.libvirt_xml.devices.hostdev.Hostdev'> object.There is no such attribute.

```

Reply:
You're right, 'reset_all' should be removed from attrs after we have its value. Thanks for finding the bug :+1:

Reply:
It works now. LGTM.
```
>>> hostdev1=hostdev.Hostdev()
>>> hostdev1.setup_attrs(**dargs)
>>> print(hostdev1)
<?xml version='1.0' encoding='UTF-8'?>
<hostdev managed="yes" mode="subsystem" type="pci"><source><address bus="0x3b" domain="0x0000" function="0x1" slot="0x00" /></source></hostdev>
>>> 
```

Reply:
Thanks again for helping test with a different xml type!